twitter/twitter.php

```diff
@@ -321,36 +317,40 @@ function twitter_settings(App $a, &$s)
-			if ($a->user['hidewall']) {
-				$s .= '<p>' . t('<strong>Note</strong>: Due your privacy settings (<em>Hide your profile details from unknown viewers?</em>) the link potentially included in public postings relayed to Twitter will lead the visitor to a blank page informing the visitor that the access to your profile has been restricted.') . '</p>';
-			}
-			$s .= '<div id="twitter-enable-wrapper">';
-			$s .= '<label id="twitter-enable-label" for="twitter-checkbox">' . t('Allow posting to Twitter') . '</label>';
-			$s .= '<input id="twitter-checkbox" type="checkbox" name="twitter-enable" value="1" ' . $checked . '/>';
-			$s .= '<div class="clear"></div>';
-			$s .= '<label id="twitter-default-label" for="twitter-default">' . t('Send public postings to Twitter by default') . '</label>';
-			$s .= '<input id="twitter-default" type="checkbox" name="twitter-default" value="1" ' . $defchecked . '/>';
-			$s .= '<div class="clear"></div>';
 
-			$s .= '<label id="twitter-mirror-label" for="twitter-mirror">' . t('Mirror all posts from twitter that are no replies') . '</label>';
-			$s .= '<input id="twitter-mirror" type="checkbox" name="twitter-mirror" value="1" ' . $mirrorchecked . '/>';
+			$field_checkbox = get_markup_template('field_checkbox.tpl');
+
+			$s .= '<div id="twitter-info" >
+				<p>' . t('Currently connected to: ') . '<a href="https://twitter.com/' . $details->screen_name . '" target="_twitter">' . $details->screen_name . '</a>
+					<button type="submit" name="twitter-disconnect" value="1">' . t('Disconnect') . '</button>
+				</p>
+				<p id="twitter-info-block">
+					<a href="https://twitter.com/' . $details->screen_name . '" target="_twitter"><img id="twitter-avatar" src="' . $details->profile_image_url . '" /></a>
+					<em>' . $details->description . '</em>
+				</p>
+			</div>';
 			$s .= '<div class="clear"></div>';
-			$s .= '</div>';
 
-			$s .= '<label id="twitter-import-label" for="twitter-import">' . t('Import the remote timeline') . '</label>';
-			$s .= '<input id="twitter-import" type="checkbox" name="twitter-import" value="1" ' . $importchecked . '/>';
-			$s .= '<div class="clear"></div>';
+			$s .= replace_macros($field_checkbox, array(
+				'$field' => array('twitter-enable', t('Allow posting to Twitter'), $enabled, t('If enabled all your <strong>public</strong> postings can be posted to the associated Twitter account. You can choose to do so by default (here) or for every posting separately in the posting options when writing the entry.'))
+			));
+			if ($a->user['hidewall']) {
+				$s .= '<p>' . t('<strong>Note</strong>: Due your privacy settings (<em>Hide your profile details from unknown viewers?</em>) the link potentially included in public postings relayed to Twitter will lead the visitor to a blank page informing the visitor that the access to your profile has been restricted.') . '</p>';
```

Comment:
Isn't a "to" missing there? "Due **to** your privacy settings ..."?

Reply:
Yep, it wasn't in the scope of this change but you're right.

Reply:
I know that it wasn't in the scope, but I'm a grammar and spelling hunter ;-)